### PR TITLE
fix(StatusBar): OS theme change no longer crashes app

### LINKED ISSUE #866: [Windows] Crash on system theme change on pages using StatusBar
## Current behavior

On pages using StatusBar, apps using Toolkit.UI crash when changing the Windows OS theme in the OS Settings:
"The application called an interface that was marshalled for a different thread. (0x8001010E (RPC_E_WRONG_THREAD))"

## Expected behavior

Apps don't crash when changing the Windows OS theme in the OS Settings

## How to reproduce it (as minimally and precisely as possible)

Start Windows sample app, change the Windows OS theme in the OS Settings while the app is open, notice crash

Nuget Package:

Package Version(s):

Affected platform(s):

- [ ] WebAssembly
- [ ] Android
- [ ] iOS
- [ ] macOS (AppKit)
- [ ] Mac Catalyst
- [ ] Skia
  - [ ] WPF
  - [ ] GTK (Linux)
  - [ ] Linux Framebuffer
  - [ ] Tizen
- [x] Windows

IDE:

- [x] Visual Studio 2022
- [ ] Visual Studio 2019
- [ ] Visual Studio Code
- [ ] Visual Studio for Mac
- [ ] Rider Windows
- [ ] Rider macOS

Relevant plugins:

### PR DESCRIPTION
**Pull request by silviuo**
GitHub Issue (If applicable): closes #866, closes unoplatform/uno.chefs#666

## PR Type

What kind of change does this PR introduce?
- Bugfix

## What is the current behavior?
On pages using StatusBar, apps using Toolkit.UI crash when changing the Windows OS theme in the OS Settings:
"The application called an interface that was marshalled for a different thread. (0x8001010E (RPC_E_WRONG_THREAD))"

## What is the new behavior?
No crash

## PR Checklist
Please check if your PR fulfills the following requirements:
- [ ] Tested code with current [supported SDKs](../README.md#supported)
- [x] Tested the changes where applicable:
	- [x] UWP
	- [x] WinUI
	- [ ] iOS
	- [x] Android
	- [x] WASM
	- [ ] MacOS
- [ ] Updated the documentation as needed:
	- [ ] [General Doc Update](https://github.com/unoplatform/uno.toolkit.ui/tree/main/doc)
	- [ ] [Controls Doc Update](https://github.com/unoplatform/uno.toolkit.ui/tree/main/doc/controls)
	- [ ] [Extensions Doc Update](https://github.com/unoplatform/uno.toolkit.ui/tree/main/doc/helpers)
	- [ ] [controls-styles.md](https://github.com/unoplatform/uno.toolkit.ui/blob/main/doc/controls-styles.md)
	- [ ] [lightweight-styling.md (LightWeight Styling Resource Keys)](https://github.com/unoplatform/uno.toolkit.ui/blob/main/doc/lightweight-styling.md)
- [ ] [Runtime Tests and/or UI Tests](https://platform.uno/docs/articles/contributing/guidelines/creating-tests.html) for the changes have been added (for bug fixes / features) (if applicable)
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal)
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

**Review by Xiaoy312 — APPROVED** [review]

**Review by kazo0 — APPROVED** [review]

File: src/Uno.Toolkit.UI/Behaviors/StatusBar.cs [hunk 1]
```diff
@@ -222,15 +222,17 @@ private static void OnUISettingsColorValuesChanged(UISettings sender, object arg
 		{
 			if (_lastActivePage is { } page)
 			{
-				if (GetSubscription(page) != null &&
+				page.GetDispatcherCompat().Schedule(() => {
```
Comment (Xiaoy312) [inline src/Uno.Toolkit.UI/Behaviors/StatusBar.cs:225]: we can probably do without scheduling on non-windows targets:
```suggestion
#if !HAS_UNO
				page.GetDispatcherCompat().Schedule(() => {
#else
				{
#endif
```

**Review by Xiaoy312 — APPROVED** [review]

**Review by agneszitte — COMMENTED** [review]
Can we add some tests for this, if possible?

File: src/Uno.Toolkit.UI/Behaviors/StatusBar.cs (hunk 1 above)
Comment (silviuo) [reply src/Uno.Toolkit.UI/Behaviors/StatusBar.cs:226]: I added Windows-specific conditionals

**Comment by silviuo** [comment]
@kazo0 I pushed the Windows conditional changes without any runtime tests

**Pull request merged**